fe/fe-core/src/main/java/com/starrocks/scheduler/mv/MVMaintenanceJob.java

```diff
@@ -0,0 +1,429 @@
+    @SerializedName("state")
+    private JobState serializedState;
+    @SerializedName("epoch")
+    private final MVEpoch epoch;
+
+    // TODO(murphy) serialize the plan, current we need to rebuild the plan for job
+    private transient ExecPlan plan;
+
+    // Runtime ephemeral state
+    // At most one thread could execute this job, this flag indicates is someone scheduling this job
+    private transient AtomicReference<JobState> state = new AtomicReference<>();
+    private transient AtomicBoolean inSchedule = new AtomicBoolean(false);
+    private transient MaterializedView view;
+    private transient ConnectContext connectContext;
+    // TODO(murphy) implement a real query coordinator
+    private transient Coordinator queryCoordinator;
+    private transient TxnBasedEpochCoordinator epochCoordinator;
+    private transient List<MVMaintenanceTask> tasks;
+
+    public MVMaintenanceJob(MaterializedView view) {
+        this.jobId = view.getId();
+        this.viewId = view.getId();
+        this.view = view;
+        this.epoch = new MVEpoch(view.getId());
+        this.serializedState = JobState.INIT;
+        this.state.set(JobState.INIT);
+        this.plan = Preconditions.checkNotNull(view.getMaintenancePlan());
+    }
+
+    public void startJob() {
+        Preconditions.checkState(state.compareAndSet(JobState.INIT, JobState.STARTED));
+    }
+
+    // TODO(murphy) make it a thread-safe with JobExecutor, what if it's under scheduling ?
+    public void stopJob() {
+        if (!inSchedule.compareAndSet(false, true)) {
+            throw UnsupportedException.unsupportedException("TODO: not support stop job running job");
+        }
+        try {
+            stopTasks();
+        } catch (Exception e) {
+            LOG.warn("stop job failed", e);
+        } finally {
+            inSchedule.compareAndSet(true, false);
+        }
+        state.set(JobState.STOPPED);
+    }
+
+    public void pauseJob() {
+        throw UnsupportedException.unsupportedException("TODO: implement pause action");
+    }
+
+    public void continueJob() {
+        throw UnsupportedException.unsupportedException("TODO: implement continue action");
+    }
+
+    public void onSchedule() throws Exception {
+        if (!inSchedule.compareAndSet(false, true)) {
+            return;
+        }
+        try {
+            switch (state.get()) {
+                case INIT:
+                    Preconditions.checkState(false, "has not started");
+                case STARTED:
+                    prepare();
+                    break;
+                case PREPARING:
+                case PAUSED:
+                case FAILED:
+                    Preconditions.checkState(false, "should not be scheduled");
+                    break;
+                case RUN_EPOCH:
+                    epoch.onSchedule();
+                    epochCoordinator.beginEpoch(epoch);
+                    epochCoordinator.commitEpoch(epoch);
+                    break;
```

Comment:
Why commitEpoch  is just after beginEpoch?  when are `TMVStartEpochTask` and  `TMVCommitEpochTask` created?

Reply:
Is that possible `beginEpoch ` <----- after some seconds/minutes --------> `commitEpoch`, the schedule time can be configured by users?

Reply:
The procedure is refined as:
1. BeginEpoch: notify all tasks to start an epoch and execution
2. WaitFinish: wait all tasks to report epoch finish or any exception happens
3. CommitEpoch: notify all tasks to commit the epoch